app/models/guide.rb

```diff
@@ -58,7 +58,7 @@ def done_for?(user)
   end
 
   def import_from_json!(json)
-    self.assign_attributes json.except('exercises', 'language', 'id_format', 'id', 'teacher_info', 'collaborators', 'private')
+    self.assign_attributes json.slice(*attribute_names).except 'id'
```

Comment:
Perhaps we could put this idiom in the `ApplicationRecord` class, so that we can do `self.slice_attribute_names(a_hash)` or `self.ẁhitelist_attributes(a_hash)`, or `self.ẁhitelist_attributes(a_hash, except: %w(type id))`